src/pgwire/src/metrics.rs

```diff
@@ -24,6 +30,12 @@ impl MetricsConfig {
                 help: "Count of completed network connections, by status",
                 var_labels: ["source", "status"],
             }),
+            time_to_first_row_seconds: registry.register(metric! {
+                name: "mz_time_to_first_row_seconds",
+                help: "Latency of an execute for a successful query from pgwire's perspective",
+                var_labels: ["source", "kind", "isolation_level"],
```

Comment:
If it'd simplify things when looking at moving this into `adapter_client.execute`, removing `kind` is fine with me. I'm not quite convinced it adds useful information.